lms/djangoapps/grades/management/commands/reset_grades.py

```diff
@@ -0,0 +1,154 @@
+            '--modified_start',
+            dest='modified_start',
+            help='Starting range for modified date (inclusive): e.g. "2016-08-23 16:43"',
+        )
+        parser.add_argument(
+            '--modified_end',
+            dest='modified_end',
+            help='Ending range for modified date (inclusive): e.g. "2016-12-23 16:43"',
+        )
+
+    def handle(self, *args, **options):
+        course_keys = None
+        modified_start = None
+        modified_end = None
+
+        run_mode = self._get_mutually_exclusive_option(options, 'delete', 'dry_run')
+        courses_mode = self._get_mutually_exclusive_option(options, 'courses', 'all_courses')
+
+        if courses_mode == 'courses':
+            try:
+                course_keys = [CourseKey.from_string(course_key_string) for course_key_string in options['courses']]
+            except InvalidKeyError:
+                raise CommandError('Invalid key specified.')
+
+        log.info("reset_grade: Started in %s mode!", run_mode)
+
+        if options.get('modified_start'):
+            modified_start = datetime.strptime(options['modified_start'], DATE_FORMAT)
+
+        if options.get('modified_end'):
+            if not modified_start:
+                raise CommandError('Optional value for modified_end provided without a value for modified_start.')
+            modified_end = datetime.strptime(options['modified_end'], DATE_FORMAT)
+
+        if options.get('dry_run'):
+            self._query_grades(PersistentSubsectionGrade, course_keys, modified_start, modified_end)
+            self._query_grades(PersistentCourseGrade, course_keys, modified_start, modified_end)
+
+        else:
+            self._delete_grades(PersistentSubsectionGrade, course_keys, modified_start, modified_end)
+            self._delete_grades(PersistentCourseGrade, course_keys, modified_start, modified_end)
+
+        log.info("reset_grade: Finished in %s mode!", run_mode)
+
+    def _delete_grades(self, grade_model_class, *args, **kwargs):
+        """
+        Deletes the requested grades in the given model, filtered by the provided args and kwargs.
+        """
+        grades_query_set = grade_model_class.query_grades(*args, **kwargs)
+        num_rows_to_delete = grades_query_set.count()
+
+        log.info("reset_grade: Deleting %s: %d row(s).", grade_model_class.__name__, num_rows_to_delete)
+
+        grade_model_class.delete_grades(*args, **kwargs)
+
+        log.info("reset_grade: Deleted %s: %d row(s).", grade_model_class.__name__, num_rows_to_delete)
+
+    def _query_grades(self, grade_model_class, *args, **kwargs):
+        """
+        Queries the requested grades in the given model, filtered by the provided args and kwargs.
+        """
+        total_for_all_courses = 0
+
+        grades_query_set = grade_model_class.query_grades(*args, **kwargs)
+        grades_stats = grades_query_set.values('course_id').order_by().annotate(total=Count('course_id'))
```

Comment:
I'm familiar with the column ordering significance of MySQL Indexes.

I'm just curious about the specific queries Django is generating there, if you can capture one for deleting across a course and for a course, I'd really like to at least look at them.  I'd like to avoid doing an accidental table scan in the future.